### PR TITLE
Guard triggering of each workflow

### PR DESCRIPTION
**Pull request by honnix**
The triggering process may fail with an unchecked exception and hence it
can affect rest of the workflows in the map.

Remediation for INCIDENT-6128.

@ulzha @fabriziodemaria PTAL

### CODE REVIEW

**Comment by fabriziodemaria** [comment]
LGTM

**Comment by honnix** [comment]
That is a a very good place to do some validation, for both schedule source and api.

File: styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java [hunk 1]
```diff
@@ -71,30 +72,30 @@ void tick() {
       }
     } catch (IOException e) {
       LOG.warn("Couldn't fetch global enabled status, skipping this run.");
-      return ;
+      return;
     }
 
-    final Map<Workflow, TriggerInstantSpec> map;
-    final Set<WorkflowId> enabled;
+    final Map<Workflow, TriggerInstantSpec> canBeTriggeredWorkflows;
```
Comment (ulzha) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: :star:

File: styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java [hunk 2]
```diff
@@ -71,30 +72,30 @@ void tick() {
...
-    map.entrySet().forEach(entry -> {
-      final Workflow workflow = entry.getKey();
-      final Schedule schedule = workflow.configuration().schedule();
-      final TriggerInstantSpec instantSpec = entry.getValue();
-
-      if (now.isBefore(instantSpec.offsetInstant())) {
-        return;
-      }
+    canBeTriggeredWorkflows.entrySet().stream()
+        .filter(entry -> !now.isBefore(entry.getValue().offsetInstant()))
```
Comment (ulzha) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: Hard to read. Can we avoid negated logic, turning it around?

Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: yeah, thought about this. that would slightly change the original logic if `it is now exactly`.

Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: That wouldn't matter. Let's change it.

Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: It was amazing we have test cases replying on exact time matching. Good catch of that!

Comment (ulzha) [inline styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java]: Ah, somewhat unfortunate, didn't intend to change the logic. Anyway fine.

**Comment by ulzha** [comment]
:+1:

**Pull request merged**